treewide: remove issue #56943 workarounds

Artturin opened the pull request:
checked with diffoscope+difflog and launching the programs

i have fixed that issue in my other gir PRs###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Artturin committed:
treewide: remove issue #56943 workarounds

checked with diffoscope+difflog and launching the programs

i have fixed that issue in my other gir PRs

Artturin committed:
dogtail: remove issue 56943 workaround

and fix double wrapping

it still fails to launch though but its not a regression

$ ./result/bin/dogtail-logout
Traceback (most recent call last):
  File "/nix/store/az8mw2pmhbwlm08176wd62hn0hfzwv8s-python3.10-dogtail-0.9.11/bin/.dogtail-logout-wrapped", line 8, in <module>
    from dogtail.tree import *
  File "/nix/store/az8mw2pmhbwlm08176wd62hn0hfzwv8s-python3.10-dogtail-0.9.11/lib/python3.10/site-packages/dogtail/tree.py", line 5, in <module>
    from dogtail import predicate
  File "/nix/store/az8mw2pmhbwlm08176wd62hn0hfzwv8s-python3.10-dogtail-0.9.11/lib/python3.10/site-packages/dogtail/predicate.py", line 6, in <module>
    from gi.repository import GLib
ModuleNotFoundError: No module named 'gi'

jtojnar commented on pkgs/applications/graphics/curtail/default.nix:
```diff
@@ -37,18 +37,16 @@ python3.pkgs.buildPythonApplication rec {
     python3.pkgs.pygobject3
-    gobject-introspection
     gettext
+    gtk3
```
Why propagate?

Artturin replied:
fix'd

jtojnar commented on pkgs/development/tools/cambalache/default.nix:
```diff
@@ -40,6 +40,8 @@ python3.pkgs.buildPythonApplication rec {
     desktop-file-utils # for update-desktop-database
     shared-mime-info # for update-mime-database
     wrapGAppsHook
+    gtk4 # checks for broadwayd and gtk4-broadwayd during configure
```
Those are used at runtime so I would expect them to be found from buildInputs.

Artturin replied:
patched

jtojnar commented on pkgs/tools/inputmethods/input-remapper/default.nix:
```diff
@@ -45,11 +45,6 @@ in
   };
 
-  # Fixes error
-  # Couldn’t recognize the image file format for file "*.svg"
```
That sounds more like gdk-pixbuf issue.

Artturin replied:
the issue was that `GDK_PIXBUF_MODULE_FILE` pointed to the gdk-pixbuf module file instead of the one in librsvg but that should be fixed since https://github.com/NixOS/nixpkgs/pull/208537

Artturin replied:
before
```
    --set 'GDK_PIXBUF_MODULE_FILE' '/nix/store/mihzapwslgszs2amsvkywb7qnhnxb5zm-gdk-pixbuf-2.42.10/lib/gdk-pixbuf-2.0/2.10.0/loaders.cache' \
```

after
```
    --set 'GDK_PIXBUF_MODULE_FILE' '/nix/store/1pkbzqhij5ay6yd7q1vj5qk4rxyi4rkc-librsvg-2.55.1/lib/gdk-pixbuf-2.0/2.10.0/loaders.cache' \
```

Artturin committed:
curtail: dont propagate unnecessary packages

pygobject3 has to be propagated for the import line in the python file
in bin

$ ./result/bin/curtail
Traceback (most recent call last):
  File "/nix/store/fqjld9nchfwqd9x1pywmjfdsqrk4lxds-curtail-1.3.1/bin/.curtail-wrapped", line 37, in <module>
    import gi
ModuleNotFoundError: No module named 'gi'

jtojnar reviewed (approved):
Looks good, thanks.

jtojnar commented:
Actually, we might want to add some tests to https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/setup-hooks/wrap-gapps-hook/default.nix

Artturin commented:
> Actually, we might want to add some tests to [`master`/pkgs/build-support/setup-hooks/wrap-gapps-hook/default.nix](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/setup-hooks/wrap-gapps-hook/default.nix?rgh-link-date=2023-02-17T19%3A03%3A41Z)

already did for gdk-pixbuf in https://github.com/NixOS/nixpkgs/pull/208537/files#diff-fa727d655e0d18c0ba8bb953613ed08fb7853f04b226341ee2fe6b17957cf523R74

for gobject-introspection there's already `typelib-x` tests written by you

jtojnar commented:
Those do not have `strictDeps = true`, though?

Artturin commented:
> Those do not have `strictDeps = true`, though?

[NixOS/nixpkgs@`8be7ab6` (#208537)](https://github.com/NixOS/nixpkgs/pull/208537/commits/8be7ab60b17f772b88358a8f1b6aeedd3306b13a)
> run tests with strictDeps

the ones using `testLib.runTest` are using strictDeps

Artturin committed:
wrapGAppsHook: move dconf.lib to depsTargetTargetPropagated

luckily the lib output does not contain binaries therefore its safe to
remove it from propagatedBuildInputs

Artturin committed:
wrapGAppsHook: run tests with strictDeps

'nix build -f . "wrapGAppsHook.tests"' pass

Artturin commented:
alright `strictDeps` enabled and `dconf.lib` moved to `depsTargetTargetPropagated`

Artturin committed:
wrapGAppsHook: add gtk3 to depsTargetTargetPropagated

for typelibs and girs

Artturin commented:
`nix build -f . "pkgsCross.aarch64-multiplatform.wrapGAppsHook.tests" --keep-going` passes other than

```
basic-contains-gdk-pixbuf> FAIL: basic-contains-gdk-pixbuf: The file “/nix/store/faxhx93d6qvizzwnw2g548pzj5sxmrwn-basic-aarch64-unknown-linux-gnu/bin/foo” should include a line containing “GDK_PIXBUF_MODULE_FILE” that also contains “/nix/store/sir0dnzpqpfl1fmx2lpv2lwjcvy17dzc-librsvg-aarch64-unknown-linux-gnu-2.55.1/lib/gdk-pixbuf-2.0/2.10.0/loaders.cache”.
```
due to https://github.com/NixOS/nixpkgs/blob/2ea8b001f6ba09cc6eb9a30f4e117cbc3ec5afe2/pkgs/development/libraries/librsvg/default.nix#L121-L126

jtojnar commented:
Would not it make more sense to keep the test failing until it is properly fixed (e.g. using emulator to generate the loaders), rather than mask the broken runtime behaviour?

Artturin committed:
librsvg: withIntrospection when stdenv.hostPlatform.emulatorAvailable buildPackages

Artturin committed:
librsvg: create loaders.cache even when cross-compiling

Artturin commented:
> Would not it make more sense to keep the test failing until it is properly fixed (e.g. using emulator to generate the loaders), rather than mask the broken runtime behaviour?

fixed it

jtojnar reviewed (approved):
Great job.

lilyinstarlight commented on pkgs/development/libraries/librsvg/default.nix:
```diff
@@ -114,10 +126,14 @@ stdenv.mkDerivation rec {
 
   # Not generated when cross compiling.
-  postInstall = lib.optionalString (stdenv.hostPlatform == stdenv.buildPlatform) ''
+  postInstall = lib.optionalString (stdenv.hostPlatform.emulatorAvailable buildPackages) ''
```
With this PR, I also need to add this diff for librsvg to build with cross again:

```diff
diff --git a/pkgs/development/libraries/librsvg/default.nix b/pkgs/development/libraries/librsvg/default.nix
index 9974826573a..87ddcd28403 100644
--- a/pkgs/development/libraries/librsvg/default.nix
+++ b/pkgs/development/libraries/librsvg/default.nix
@@ -147,11 +147,11 @@ stdenv.mkDerivation (finalAttrs: {
     mv $GDK_PIXBUF/loaders.cache.tmp $GDK_PIXBUF/loaders.cache
 
     mkdir -p "$out/share/bash-completion/completions/"
-    $out/bin/rsvg-convert --completion bash > "$out/share/bash-completion/completions/rsvg-convert"
+    ${stdenv.hostPlatform.emulator buildPackages} $out/bin/rsvg-convert --completion bash > "$out/share/bash-completion/completions/rsvg-convert"
     mkdir -p "$out/share/zsh/site-functions/"
-    $out/bin/rsvg-convert --completion zsh > "$out/share/zsh/site-functions/_rsvg-convert"
+    ${stdenv.hostPlatform.emulator buildPackages} $out/bin/rsvg-convert --completion zsh > "$out/share/zsh/site-functions/_rsvg-convert"
     mkdir -p "$out/share/fish/vendor_completions.d/"
-    $out/bin/rsvg-convert --completion fish > "$out/share/fish/vendor_completions.d/rsvg-convert.fish"
+    ${stdenv.hostPlatform.emulator buildPackages} $out/bin/rsvg-convert --completion fish > "$out/share/fish/vendor_completions.d/rsvg-convert.fish"
   '';
 
   postFixup = lib.optionalString withIntrospection ''
```

Is this the correct approach for fixing that?

If so, I'll open a PR to fix librsvg cross (I'll probably use a `let rsvg-convert = "${stdenv.hostPlatform.emulator buildPackages} $out/bin/rsvg-convert" in ...` to shorten it a little, though)

jtojnar replied:
We already have that on staging: https://github.com/NixOS/nixpkgs/pull/244602

lilyinstarlight replied:
Ah, thanks! I did a quick search but missed that. Sorry for the noise 😅
